Homework9/Homework9/Database.cs

```diff
@@ -0,0 +1,175 @@
+                    select orders).ToArray();
+        }
+
+        public Order GetOrder(int orderId)
+        {
+            return Orders.SingleOrDefault(o => o.Id == orderId);
+        }
+
+        public decimal GetMoneySpentBy(int customerId)
+        {
+            var productsPrice = from order in Orders
+                                join products in Products on order.ProductId equals products.Id
+                                where order.CustomerId == customerId
+                                select new
+                                {
+                                    Price = products.Price
+                                };
+
+            return productsPrice.Sum(n => n.Price);
+        }
+
+        public Product[] GetAllProductsPurchased(int customerId)
+        {
+            return (from orders in Orders
+                    join products in Products on orders.ProductId equals products.Id
+                    where orders.CustomerId == customerId
+                    select products).ToArray();
+        }
+
+        public Product[] GetUniqueProductsPurchased(int customerId)
+        {   
+            return (from orders in Orders 
+                    join products in Products on orders.ProductId equals products.Id 
+                    where orders.CustomerId == customerId 
+                    select products).ToArray();
+        }
+
+        public bool HasEverPurchasedProduct(int customerId, int productId)
+        {
+            return Orders.Any(o => o.ProductId == productId && o.CustomerId == customerId);
+        }
+
+        public bool AreAllPurchasesHigherThan(int customerId, decimal targetPrice)
+        {
+            var newTable = Orders.Where(o => o.CustomerId == customerId).Join(Products,
+                 o => o.ProductId,
+                 p => p.Id,
+                 (o, p) => new
+                 {
+                     Customer = o.CustomerId,
+                     Price = p.Price
+                 });
+
+            return newTable.All(n => n.Price > targetPrice);
+        }
+
+        public bool DidPurchaseAllProducts(int customerId, params int[] productIds)
+        {
+            var selectedProducts = (from orders in Orders
+                        where orders.CustomerId == customerId
+                        select orders.ProductId).ToArray();
+
+            return selectedProducts.Any(p => productIds.Contains(p));
+        }
+
+        public int GetTotalProductsPurchased(int productId)
+        { 
+            return Orders.Where(o => o.ProductId == productId).Count();
+        }
+
+        public CustomerOverView GetCustomerOverview(int customerId)
+        {
+            var name = (from customers in Customers
+                       where customers.Id == customerId
+                       select customers.Name).First();
+
+            var totalProductsPurchased = Orders.Where(o => o.CustomerId == customerId).Count();
+
+            var allProducts = from orders in Orders
```

Comment:
в имя тaкой переменной лучше еще писать по чем группируем, а то дальше непонятно что происходит
```suggestion
            var allProductsByName = from orders in Orders
```